syntax = "proto3";

import "kv.proto";

package externalexecutor.protobuf;

option optimize_for = LITE_RUNTIME;

service Historical
{
  // Historical data look up
  rpc GetHistoricalData(HistoricalData) returns (QueryResponse) {}
}

message HistoricalData
{
  string map_name = 1;
  bytes key = 2;
  int32 view = 3;
  int32 seqno = 4;
}

message RetryQuery { bool retryquery = 1; }
message QueryResponse
{
  externalexecutor.protobuf.KVValue data = 1;
  RetryQuery retry = 2;
}